rero_ils/modules/acq_accounts/api.py

```diff
@@ -61,49 +64,333 @@ class AcqAccount(IlsRecord):
+
+        query.aggs.metric('total_amount', 'sum', field='total_amount')
+        results = query.execute()
+        self_amount = results.aggregations.total_amount.value
+
+        # Encumbrance of children accounts
+        query = AcqAccountsSearch().filter('term', parent__pid=self.pid)
+        query.aggs.metric('total', 'sum', field='encumbrance_amount.total')
+        results = query.execute()
+        children_amount = results.aggregations.total.value
+
+        return self_amount, children_amount
+
+    @property
+    def expenditure_amount(self):
+        """Get the expenditure amount related to this account.
+
+        This amount is the sum of :
+          * Invoice lines related to this account depending of invoice lines
+            status ; only "received" invoice lines must be used to compute this
+            amount.
+          * expenditure of all children account
+
+        @:return A tuple of expenditure amount : First element if expenditure
+                 for this account, second element is the children expenditure.
+        """
+        # TODO :: compute the expenditure based on acq_invoices.
+
+        # Expenditure of children accounts
+        query = AcqAccountsSearch().filter('term', parent__pid=self.pid)
+        query.aggs.metric('total', 'sum', field='expenditure_amount.total')
+        results = query.execute()
+        children_amount = results.aggregations.total.value
+        return 0, children_amount
+
+    @property
+    def remaining_balance(self):
+        """Get the remaining balances for this account.
+
+        The balance is computed from the initial account amount reduces by
+        encumbrance and expenditure of myself + distribution to children
+        accounts.
+        The total balance for this accout is computed from the initial account
+        reduces by the sum of (self encumbrance + children encumbrance) and sum
+        of (self expenditure + children expenditure).
+
+        :return: A tuple with self balance and total balance
+        """
+        initial_amount = self.get('allocated_amount')
+        encumbrance = self.encumbrance_amount
+        expenditure = self.expenditure_amount
+
+        self_balance = initial_amount \
+            - self.distribution \
+            - encumbrance[0] \
+            - expenditure[0]
+        total_balance = initial_amount \
+            - sum(list(self.encumbrance_amount)) \
+            - sum(list(self.expenditure_amount))
+
+        return self_balance, total_balance
+
+    @property
+    def distribution(self):
+        """Get the amount distribute into children account.
+
+        The distribution is the sum of direct children account initial amount.
+        If this account has 2 children : AccountA and AccountB with both an
+        allocated amount of 2000, then the distribution of the parent account
+        is 4000.
+        The distribution cannot exceed the account allocated amount.
```

Comment:
```suggestion
        If this account has 2 children, 'AccountA' and 'AccountB', with both an
        allocated amount of 2000, then the amount to be distributed by the parent account
        is 4000.
        The distribution cannot exceed the allocated amount of the account.
```
I'm not sure about my *amount to be distributed by* instead of *distribution*, but I'm not sure either that *distribution of the parent account* means something.